pkg/staking/stakingcontract/contract_test.go

```diff
@@ -0,0 +1,539 @@
+	bzzTokenAddress := common.HexToAddress("eeee")
+	nonce := common.BytesToHash(make([]byte, 32))
+	addr := swarm.MustParseHexAddress("f30c0aa7e9e2a0ef4c9b1b750ebfeaeb7c7c24da700bb089da19a46e3677824b")
+
+	t.Run("ok", func(t *testing.T) {
+		t.Parallel()
+
+		prevStake := big.NewInt(0)
+		expectedCallData, err := stakingcontract.StakingABI.Pack("stakeOfOverlay", common.BytesToHash(addr.Bytes()))
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		contract := stakingcontract.New(owner, stakingAddress, bzzTokenAddress,
+			transactionMock.New(
+				transactionMock.WithCallFunc(func(ctx context.Context, request *transaction.TxRequest) (result []byte, err error) {
+					if *request.To == stakingAddress {
+						if !bytes.Equal(expectedCallData[:64], request.Data[:64]) {
+							return nil, fmt.Errorf("got wrong call data. wanted %x, got %x", expectedCallData, request.Data)
+						}
+					}
+					if *request.To == stakingAddress {
+						return prevStake.FillBytes(make([]byte, 32)), nil
+					}
+					return nil, errors.New("unexpected call")
+				})),
+			nonce)
+
+		stakedAmount, err := contract.GetStake(ctx, addr)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if stakedAmount.Cmp(big.NewInt(1)) == 0 {
+			t.Fatalf("expected %v got %v", big.NewInt(1), stakedAmount)
+		}
+	})
+
+	t.Run("error with unpacking", func(t *testing.T) {
+		t.Parallel()
+		expectedCallData, err := stakingcontract.StakingABI.Pack("stakeOfOverlay", common.BytesToHash(addr.Bytes()))
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		contract := stakingcontract.New(owner, stakingAddress, bzzTokenAddress,
+			transactionMock.New(
+				transactionMock.WithCallFunc(func(ctx context.Context, request *transaction.TxRequest) (result []byte, err error) {
+					if *request.To == stakingAddress {
+						if !bytes.Equal(expectedCallData[:64], request.Data[:64]) {
+							return nil, fmt.Errorf("got wrong call data. wanted %x, got %x", expectedCallData, request.Data)
+						}
+						return []byte{}, nil
+					}
+					return nil, errors.New("unexpected call")
+				})),
+			nonce)
+
+		_, err = contract.GetStake(ctx, addr)
+		if err == nil {
+			t.Fatal("expected error with unpacking")
+		}
+	})
+
+	t.Run("with invalid call data", func(t *testing.T) {
+		t.Parallel()
+
+		prevStake := big.NewInt(0)
+		expectedCallData, err := stakingcontract.StakingABI.Pack("stakeOfOverlay", common.BytesToHash(owner.Bytes()))
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		contract := stakingcontract.New(owner, stakingAddress, bzzTokenAddress,
+			transactionMock.New(
+				transactionMock.WithCallFunc(func(ctx context.Context, request *transaction.TxRequest) (result []byte, err error) {
+					if *request.To == stakingAddress {
+						if !bytes.Equal(expectedCallData[:64], request.Data[:64]) {
+							return nil, fmt.Errorf("got wrong call data. wanted %x, got %x", expectedCallData, request.Data)
+						}
+					}
+					if *request.To == stakingAddress {
+						return prevStake.FillBytes(make([]byte, 32)), nil
+					}
```

Comment:
could these two `if` statements combined to single?

Reply:
yes lemme do this